### PR TITLE
Make a "reset.sh" script

### LINKED ISSUE #269: Make a "reset.sh" script.
**Branch:** `Feature` (osx-0.16.x os x installers)

This is the equivalent of the "Reset App" button of the .app.

- [x] delete the symlinked /usr/bin/kalite or /usr/local/bin/kalite
- [x] delete the /Library/LaunchAgents/org.learningequality.kalite.plist
- [x] Remove the KALITE_PYTHON env var

### PR DESCRIPTION
**Pull request by amodia**
@cpauya this fixes #269

### CODE REVIEW

File: osx/reset.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+#  This will remove the following.
+#    1. Unset environment variable: KALITE_PYTHON
+#    2. remove the .plist file, need admin
+#    3. delete the symlinked /usr/bin/kalite command, need admin
+
+KALITE_PLIST="/tmp/org.learningequality.kalite.plist"
+KALITE_PLIST_LIBRARY="/Library/LaunchAgents/org.learningequality.kalite.plist"
```
Comment (cpauya) [inline osx/reset.sh]: Can you put "org.learningequality.kalite.plist" as one variable that we can re-use?  It'll be easier to maintain later.

File: osx/reset.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+KALITE_EXECUTABLE_PATH="$(which kalite)"
+
+
+# Delete the .plist file.
+if [ -f $KALITE_PLIST ]
+then
```
Comment (cpauya) [inline osx/reset.sh]: Let's make our bash scripts consistent by using:

```bash
if [ boolean-statement ]; then
... blah
else
... blah
fi
```
Do the same with the others.

File: osx/reset.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+if [ -f "$LIBRARY/$KALITE_PLIST" ]; then
+    echo "Removing $LIBRARY/$KALITE_PLIST "
+    sudo rm "$LIBRARY/$KALITE_PLIST"
+else
+    echo "$LIBRARY/$KALITE_PLIST not found."
+fi
+
+# Delete the symlinked `kalite` executable.
+# REF: http://stackoverflow.com/questions/21799441/test-if-executable-exists-in-unix
+if which kalite > /dev/null 2>&1; then
```
Comment (cpauya) [inline osx/reset.sh]: @amodia Can you re-use the `KALITE_EXECUTABLE_PATH` variable you declared above?

File: osx/reset.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    sudo rm "$TEMP/$KALITE_PLIST"
+else
+    echo "$KALITE_PLIST not found. "
+fi
+
+if [ -f "$LIBRARY/$KALITE_PLIST" ]; then
+    echo "Removing $LIBRARY/$KALITE_PLIST "
+    sudo rm "$LIBRARY/$KALITE_PLIST"
+else
+    echo "$LIBRARY/$KALITE_PLIST not found."
```
Comment (cpauya) [inline osx/reset.sh]: Create a variable to hold these four (4) instances of the same string concatenation: `$LIBRARY/$KALITE_PLIST` and use that instead.

File: osx/reset.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+LIBRARY="/Library/LaunchAgents"
+KALITE_EXECUTABLE_PATH="$(which kalite)"
+
+
+# Delete the .plist file.
+if [ -f "$TEMP/$KALITE_PLIST" ]; then
+    echo "Removing $TEMP/$KALITE_PLIST "
+    sudo rm "$TEMP/$KALITE_PLIST"
+else
+    echo "$KALITE_PLIST not found. "
```
Comment (cpauya) [inline osx/reset.sh]: Create a variable to hold these four (4) instances of the same string concatenation: `$TEMP/$KALITE_PLIST` and use that instead.

**Comment by djallado** [comment]
@amodia, I tested this PR, when I run `reset.sh`, It deletes `KALITE_PYTHON` env var, `/Library/LaunchAgents/org.learningequality.kalite.plist`, and the symlink of `kalite` executable in `/usr/bin`.

**Comment by cpauya** [comment]
Nice one @djallado - merging.

**Pull request merged**